starter-code/index.js

```diff
@@ -1,11 +1,46 @@
 class SortedList {
-  constructor() {}
-  add(item) {}
-  get(pos) {}
-  max() {}
-  min() {}
-  average() {}
-  sum() {}
+  constructor(items, length) {
+    this.items = [];
+    this.length = this.items.length;
+  }
+  add(item) {
+    this.items.push(item);
+    this.items.sort((a , b) => a-b);
+    this.length=this.items.length;
+  }
+  get(pos) {
+    return this.items[pos];
+  }
+  max() {
+    if(this.items.length===0) {
+      throw new Error("EmptySortedList");
+    } else {
+      return Math.max(...this.items);
+    }
+  }
+  min() {
+    if(this.items.length===0) {
+      throw new Error("EmptySortedList");
+    } else {
+      return Math.min(...this.items);
+    }
+
+  }
+  average() {
+    if(this.items.length===0) {
+      throw new Error("EmptySortedList");
+    } else {
+      return this.items.reduce((acu, it) => acu + it) / this.items.length;
+    }
+  }
+  sum() {
+    if(this.items.length===0){
+      return 0;
+    } else {
+     return this.items.reduce((acu, it) => acu + it);
```

Comment:
Importante colocar no reduce o segundo argumento que é o valor inicial _arr.reduce(callback[, initialValue])_, neste caso seria '0'.  _this.items.reduce((acu, it) => acu + it, 0)_
_